### PR TITLE
searxng: unstable-2023-06-26 -> unstable-2023-07-19, freeze version

### PR DESCRIPTION
**Pull request by 999eagle**
###### Description of changes

Upstream changes: https://github.com/searxng/searxng/compare/da7c30291dcf53cc5b3d98f9aada5615cd1593a9..a446dea1bb492eac417de9a900fae7cdf94aeec0

This also includes the exact version used to build SearXNG in the final package, see https://github.com/searxng/searxng/issues/2563#issuecomment-1643602674.

Deployed to https://sx.catgirl.cloud with https://git.catgirl.cloud/999eagle/dotfiles-nix/-/commit/64024c59839358fe04a187a5c03fdef3c646b081.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/web-apps/searxng/default.nix [hunk 1]
```diff
@@ -3,24 +3,40 @@
...
+    let
+      versionString = lib.concatStringsSep "." (builtins.tail (lib.splitString "-" version));
+      commitAbbrev = builtins.substring 0 8 commit;
+    in
+    ''
+      export SEARX_DEBUG="true";
+
+      cat > searx/version_frozen.py <<EOF
+      VERSION_STRING="${versionString}+${commitAbbrev}"
+      VERSION_TAG="${versionString}+${commitAbbrev}"
```
Comment (return42) [inline pkgs/servers/web-apps/searxng/default.nix:34]: Is `${versionString}` replaced by the nixos package version (`unstable-2023-07-19` from above)?

If so you might run into trouble with [PEP-440 Version scheme](https://peps.python.org/pep-0440/#version-scheme) .. in the past we (SearXNG team) we had some trouble with the version scheme accepted by pip and other versioning tools (that implement strict PEP-440' versions scheme).

- https://github.com/searxng/searxng/pull/2122
- https://github.com/searxng/searxng/pull/2117

Comment (999eagle) [reply pkgs/servers/web-apps/searxng/default.nix:34]: `versionString` is calculated from the package version in line 26, so `versionString = lib.concatStringsSep "." (builtins.tail (lib.splitString "-" version))`. Currently, this results in `2023.07.19`.

I don't think PEP-440 impacts the nix package at all as versioning is entirely handled through nix (which only cares about the package version). The `VERSION_STRING` written to `version_frozen.py` is only used for displaying in the web interface as far as I can tell.

File: pkgs/servers/web-apps/searxng/default.nix [hunk 2]
```diff
@@ -3,24 +3,40 @@
 , fetchFromGitHub
 }:
 
+let
+  commit = "a446dea1bb492eac417de9a900fae7cdf94aeec0";
+in
```
Comment (SuperSandro2000) [inline pkgs/servers/web-apps/searxng/default.nix:8]:
```suggestion
```

File: pkgs/servers/web-apps/searxng/default.nix [hunk 3]
```diff
@@ -3,24 +3,40 @@
...
   pname = "searxng";
-  version = "unstable-2023-06-26";
+  version = "unstable-2023-07-19";
 
   src = fetchFromGitHub {
     owner = pname;
     repo = pname;
-    rev = "da7c30291dcf53cc5b3d98f9aada5615cd1593a9";
-    sha256 = "sha256-kbNw/YgcBZNkmn2nmsnEnc9Y8MJg3zGFdW1x9GIo+dM=";
+    rev = commit;
```
Comment (SuperSandro2000) [inline pkgs/servers/web-apps/searxng/default.nix:16]:
```suggestion
    rev = "a446dea1bb492eac417de9a900fae7cdf94aeec0";
```

File: pkgs/servers/web-apps/searxng/default.nix [hunk 4]
```diff
@@ -3,24 +3,40 @@
...
     sed -i 's/==.*$//' requirements.txt
   '';
 
-  preBuild = ''
-    export SEARX_DEBUG="true";
-  '';
+  preBuild =
+    let
+      versionString = lib.concatStringsSep "." (builtins.tail (lib.splitString "-" version));
+      commitAbbrev = builtins.substring 0 8 commit;
```
Comment (SuperSandro2000) [inline pkgs/servers/web-apps/searxng/default.nix:27]:
```suggestion
      commitAbbrev = builtins.substring 0 8 src.rev;
```

File: pkgs/servers/web-apps/searxng/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/servers/web-apps/searxng/default.nix:34]: That should be fine, as long as pip isn't complaining. If we need more information than currently available we should just fill them with dummy values to keep things simple.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
image

**Pull request merged**